File: src/main/java/bad/robot/radiate/teamcity/AllProjectsAsSingleTask.java
package bad.robot.radiate.teamcity;

import bad.robot.radiate.monitor.MonitoringTask;
import bad.robot.radiate.monitor.MonitoringTasksFactory;
import bad.robot.radiate.monitor.ThreadSafeObservable;

import java.util.Arrays;
import java.util.List;

/** @see {@link bad.robot.radiate.monitor.MonitoringTasksFactory#singleAggregate()} */
public class AllProjectsAsSingleTask extends ThreadSafeObservable implements MonitoringTasksFactory {

    @Override
    public List<MonitoringTask> create() {
        TeamCityConfiguration configuration = YmlConfiguration.loadOrCreate(new BootstrapTeamCity(), this);
        return Arrays.<MonitoringTask>asList(new AllProjectsMonitor(configuration));
    }

    @Override
    public String toString() {
        return "multiple projects as a single aggregate";
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/AllProjectsAsSingleTask.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.monitor.{ThreadSafeObservable, MonitoringTask, MonitoringTasksFactory}
import bad.robot.radiate.Error
import scalaz.\/
import scalaz.syntax.either._

class AllProjectsAsSingleTask extends ThreadSafeObservable with MonitoringTasksFactory {
  def create: Error \/ List[MonitoringTask] = {
    val configuration = YmlConfiguration.loadOrCreate(new BootstrapTeamCity, this)
    List(new AllProjectsMonitor(configuration)).right
  }

  override def toString = "multiple projects as a single aggregate"
}